src/data/repositories/DataSetD2Api.ts

```diff
@@ -33,40 +33,85 @@ export class DataSetD2Api {
             const attributesToFilter = options.filters.projectsIds
                 ? [attributes.project.id]
                 : [attributes.createdByApp.id];
-            return apiToFuture(
-                this.api.models.dataSets.get({
-                    pageSize: options.paging.pageSize,
-                    page: options.paging.page,
-                    filter: {
-                        "attributeValues.attribute.id": { in: attributesToFilter },
-                        "attributeValues.value": this.isProjectFilterPresent(options)
-                            ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
-                            : { eq: "true" },
-                        identifiable: { token: options.filters.search },
-                        id: { in: options.filters.ids },
-                    },
-                    fields: {
-                        ...dataSetFields,
-                        ...(options.includeOrgUnits
-                            ? {
-                                  organisationUnits: {
-                                      id: true,
-                                      path: true,
-                                      displayName: true,
-                                  },
-                              }
-                            : {}),
+            return this.getDataSets(options, attributes, coreCompetencies, attributesToFilter);
+        });
+    }
+
+    getWithOrgUnits(options: D2DataSetOptions): FutureData<Paginated<DataSet>> {
+        return this.getBaseData().flatMap(({ attributes, coreCompetencies }) => {
+            return this.getDataSetsWithOrgUnits(options, attributes, coreCompetencies, [
+                attributes.createdByApp.id,
+            ]);
+        });
+    }
+
+    private getDataSets(
+        options: D2DataSetOptions,
+        attributes: D2Config["attributes"],
+        coreCompetencies: CoreCompetency[],
+        attributesToFilter: Id[]
+    ): FutureData<Paginated<DataSet>> {
+        return apiToFuture(
+            this.api.models.dataSets.get({
+                pageSize: options.paging.pageSize,
+                page: options.paging.page,
+                filter: {
+                    "attributeValues.attribute.id": { in: attributesToFilter },
+                    "attributeValues.value": options.filters.projectsIds
+                        ? { in: [...this.getOrThrow(options.filters.projectsIds)] }
+                        : { eq: "true" },
```

Comment:
This looks confusing to me. We have a filter id/value (that's a problem with DHIS2, as we already saw), and the filter is used to both filter by projectIds and something boolean (I guess, createdByApp). Not sure the best approach, but I think this needs some refactoring.